src/etc/inc/plugins.inc.d/captiveportal.inc

```diff
@@ -0,0 +1,190 @@
+<?php
+
+/*
+ * Copyright (C) 2025 Deciso B.V.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED ``AS IS'' AND ANY EXPRESS OR IMPLIED WARRANTIES,
+ * INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+ * AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+ * OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+function captiveportal_services()
+{
+    $services = [];
+
+    $cp = new \OPNsense\CaptivePortal\CaptivePortal();
+    if ($cp->isEnabled()) {
+        $services[] = array(
+            'pidfile' => '/var/run/lighttpd-api-dispatcher.pid',
+            'description' => gettext('Captive Portal'),
+            'configd' => array(
+                'restart' => array('captiveportal restart'),
+                'start' => array('captiveportal start'),
+                'stop' => array('captiveportal stop'),
+            ),
+            'name' => 'captiveportal',
+        );
+    }
+
+    return $services;
+}
+
+function captiveportal_cron()
+{
+    global $config;
+
+    $jobs = [];
+
+    if (!empty($config['system']['captiveportalbackup']) && $config['system']['captiveportalbackup'] > 0) {
+        $jobs[]['autocron'] = array(
+            '/usr/local/etc/rc.syshook.d/backup/20-captiveportal',
+            '0',
+            '*/' . $config['system']['captiveportalbackup']
+        );
+    }
+
+    return $jobs;
+}
+
+function captiveportal_syslog()
+{
+    $logfacilities = [];
+
+    $logfacilities['portalauth'] = ['facility' => ['captiveportal']];
+
+    return $logfacilities;
+}
+
+function captiveportal_firewall($fw)
+{
+    global $config;
+
+    $cp = new \OPNsense\CaptivePortal\CaptivePortal();
+    if ($cp->isEnabled()) {
+        foreach ($cp->zones->zone->iterateItems() as $zone) {
+            if (empty((string)$zone->enabled)) {
```

Comment:
```suggestion
            if ($zone->enabled->isEmpty()) {
```

we added these recently, practical to use in these conditions

Reply:
neat :)